ipa-core/src/config.rs

```diff
@@ -117,6 +119,88 @@ impl<F: ConnectionFlavor> NetworkConfig<F> {
     }
 }
 
+/// Reads a the config for a specific, single, sharded server from string. Expects config to be
+/// toml format. The server in the network is specified via `id`, `shard_index` and
+/// `shard_count`.
+///
+/// First we read the configuration without assigning any identities. The number of peers in the
+/// configuration must be a multiple of 6, or 3 as a special case to support older, non-sharded
+/// configurations.
+///
+/// If there are 3 entries, we assign helper identities for them. We create a dummy sharded
```

Comment:
I think these file conventions are important since they are quite sticky. I will create a separate struct just for that serialization and keep NetworkConfig as is. Let me know what you think